m4opt/models/math.py

```diff
@@ -0,0 +1,247 @@
+    >>> _get_intervals(models.Exponential1D(tau=1*u.m))
+    (<Quantity -inf m>,<Quantity inf m>)
+
+    Compound models are supported too. Supported operations include function
+    composition (`|`):
+
+    >>> _get_intervals(models.Box1D(width=3) | models.Shift(offset=2))
+    (0.5,3.5)
+    >>> _get_intervals(models.Box1D(width=3*u.m) | models.Shift(offset=2*u.m))
+    (<Quantity 0.5 m>,<Quantity 3.5 m>)
+
+    Multiplication (`*`) results in an intersection:
+
+    >>> model1 = models.Box1D(x_0=-0.5, width=3)
+    >>> model2 = models.Box1D(x_0=0.5, width=3)
+    >>> _get_intervals(model1)
+    (-2.0,1.0)
+    >>> _get_intervals(model2)
+    (-1.0,2.0)
+    >>> _get_intervals(model1 * model2)
+    (-1.0,1.0)
+    >>> model1 = models.Box1D(x_0=-0.5*u.m, width=3*u.m)
+    >>> model2 = models.Box1D(x_0=0.5*u.m, width=3*u.m)
+    >>> _get_intervals(model1 * model2)
+    (<Quantity -1. m>,<Quantity 1. m>)
+
+    Addition (`+`) and subtraction (`-`) result in a union, but holes are kept.
+
+    >>> model1 = models.Box1D(x_0=-0.5, width=3)
+    >>> model2 = models.Box1D(x_0=0.5, width=3)
+    >>> _get_intervals(model1 + model2)
+    (-2.0,-1.0) | (-1.0,2.0)
+    >>> _get_intervals(model1 - model2)
+    (-2.0,-1.0) | (-1.0,2.0)
+    >>> model1 = models.Box1D(x_0=-0.5*u.m, width=3*u.m)
+    >>> model2 = models.Box1D(x_0=0.5*u.m, width=3*u.m)
+    >>> _get_intervals(model1 + model2)
+    (<Quantity -2. m>,<Quantity -1. m>) | (<Quantity -1. m>,<Quantity 2. m>)
+    >>> _get_intervals(model1 - model2)
+    (<Quantity -2. m>,<Quantity -1. m>) | (<Quantity -1. m>,<Quantity 2. m>)
+    """
+    if isinstance(model, CompoundModel):
+        if model.op == '|':
+            return _get_intervals(model.left).apply(_map_interval(model.right))
+        elif model.op == '*':
+            return _get_intervals(model.left) & _get_intervals(model.right)
+        elif model.op in {'+', '-', '/'}:
+            lhs = portion.IntervalDict({_get_intervals(model.left): 'lhs'})
+            rhs = portion.IntervalDict({_get_intervals(model.right): 'rhs'})
+            union = lhs | rhs
+            return portion.Interval(
+                *(intervals.apply(_to_open_interval) for intervals in union))
+        else:
+            raise NotImplementedError(f'operation {model.op} not supported')
+    else:
+        try:
+            bbox = model.bounding_box
+        except NotImplementedError:
+            unit = _get_1d_units_for_dict(model.input_units)
+            return portion.open(-np.inf * unit, np.inf * unit)
+        else:
+            (lo, hi), = bbox
+            return portion.open(lo, hi)
+
+
+def integrate(model: Model, *,
+              quick_and_dirty_npts: Optional[int] = None,
+              **kwargs) -> Union[Quantity, float]:
+    """Integrate a 1D model using adaptive trapezoidal quadrature.
+
+    Parameters
+    ----------
+    model : :class:`astropy.modeling.Model`
+        The model to integrate.
+    quick_and_dirty_npts : int, optional
```

Comment:
maybe
```suggestion
    non_adaptive_npts : int, optional
```
or something like that? "quick and dirty" is in some ways accurate, but it took me some time to understand what you really meant here algorithmically

Reply:
Actually I prefer "quick and dirty" because it forces the user to accept that there are no guarantees of accuracy.

If we keep it called `quick_and_dirty_npts`, can we improve the documentation to make it more clear what it does differently?